phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -1306,6 +1306,12 @@ private TableDescriptor ensureTableCreated(byte[] physicalTableName, PTableType
                       .build().buildException();
                 }
             }
+            // If DoNotUpgrade config is set only check namespace mapping and
+            // Client-server compatibility for system tables.
+            if(isDoNotUpgradePropSet && SchemaUtil.isSystemTable(physicalTableName)) {
+                checkClientServerCompatibility(SYSTEM_CATALOG_NAME_BYTES);
```

Comment:
If the client has `doNotUpgrade == true` and SYSTEM.CATALOG doesn't exist, `checkClientServerCompatibility` will throw a `TableNotFoundException`. If we were to catch this exception and rethrow as `UpgradeRequiredException`, won't it follow a similar path as expected today?